Could not create session because of error: [object Object]

Sorry I didn't check-in with you sooner @arikon - https://github.com/selenodium/selenodium-grid/pull/14 did indeed fix the node registration issue I was experiencing! I've been fiddling around since then trying to run a very basic test against a browser via selenodium:

```python
from selenium import webdriver
from selenium.webdriver.common.keys import Keys
from selenium.webdriver.common.desired_capabilities import DesiredCapabilities

driver = webdriver.Remote(
    command_executor='http://localhost:4444/wd/hub',
    desired_capabilities=DesiredCapabilities.CHROME)
driver.get('http://www.google.com')
html = driver.page_source
print (html)
driver.close()
```
after having registered a node using Selenium's standalone server. However it seems that selenodium is unable to create a new session after placing a request to the `requestServlet`. I traced the session creation calls from there (`Registry.js` -> `SessionQueue.js` -> `NodeSet.js` -> `Node.js` -> `NodeSlot.js`) which gets me to this piece of code:

```javascript
getNewSession: function(req) {
        var self = this;
        if (self.isLocked()) {
            log.debug('Slot on node with ID=%s is locked, skip', self.node.getId());
            return q(null);
        }

        // match capabilities
        var caps = req.capabilities;
        if (!self.hasCapability(caps)) {
            log.debug('Capabilities for slot on node with ID=%s do not match', self.node.getId());
            log.debug('* desired capabilities: %j', caps);
            log.debug('* slot capabilities: %j', self.caps);
            return q(null);
        }

        this.lock();

        req = extend({retries: 2}, req);
        return self.getDriver().getNewSessionFromNode(req, self.node)
            .catch(function(err) {
                // unlock node on session opening error
                self.unlock();
                return q.reject(err);
            })
            .then(function(res) {
                // save session on success
                self.session = new Session(self, res.sessionId, res.capabilities, res.desiredCapabilities);
                return self.session;
            });
    },
```
Now it appears the session declaration and allocation is never taken, which leads me to believe that `getNewSessionFromNode` is not fulfilled. That function is called in `CommonImpl.js`:
```javascript
getNewSessionFromNode: function(req, node) {
        var self = this,
            caps = self.getRequestCapabilities(req);

        // send open session request
        return self.proxyRequestToNode(req, node)
            .then(function(res) {
                // process response
                console.log('hello');
                return self.parseProxyResponse(res);
            })
            .spread(function(sessionId, resCaps) {
                return {
                    sessionId: sessionId,
                    capabilities: resCaps,
                    desiredCapabilities: caps
                }
            })
            .catch(function(err) {
                log.debug('Error ocured during session opening on node with ID=%s', node.getId());
                log.debug(err.stack || err);
                return q.reject(err);
            });
    },
```
And again, here it looks like it's `self.proxyRequestToNode(req, node)` that isn't being fulfilled. But the only thing that this function does is call `apps.processJsonBody` from `http-apps.js`, and nothing something to be wrong there, as it returns exactly what it should:

```
{ version: [ 1, 1 ],
  method: 'GET',
  path: '/grid/api/proxy?id=http://10.211.101.57:5555',
  _pathInfo: null,
  scriptName: '',
  scheme: 'http',
  hostname: '10.211.101.57',
  port: 4444,
  host: '10.211.101.57:4444',
  remoteHost: '::ffff:10.211.101.57',
  remotePort: 54457,
  url: 'http://10.211.101.57:4444/grid/api/proxy?id=http://10.211.101.57:5555',
  body: 
   { read: [Function],
     forEach: [Function],
     close: [Function],
     node: 
      { _readableState: [Object],
        readable: false,
        domain: null,
        _events: [Object],
        _maxListeners: 10,
        socket: [Object],
        connection: [Object],
        httpVersion: '1.1',
        complete: true,
        headers: [Object],
        trailers: {},
        _pendings: [],
        _pendingIndex: 0,
        url: '/grid/api/proxy?id=http://10.211.101.57:5555',
        method: 'GET',
        statusCode: null,
        client: [Object],
        _consuming: true,
        _dumped: false,
        httpVersionMajor: 1,
        httpVersionMinor: 1,
        upgrade: false,
        pipe: [Function],
        addListener: [Function],
        on: [Function],
        pause: [Function],
        resume: [Function],
        read: [Function] } },
  headers: 
   { host: '10.211.101.57:4444',
     connection: 'Keep-Alive',
     'user-agent': 'Apache-HttpClient/4.3.6 (java 1.5)',
     'accept-encoding': 'gzip,deflate' },
  node: 
   { _readableState: 
      { highWaterMark: 16384,
        buffer: [],
        length: 0,
        pipes: null,
        pipesCount: 0,
        flowing: false,
        ended: true,
        endEmitted: true,
        reading: false,
        calledRead: true,
        sync: false,
        needReadable: true,
        emittedReadable: false,
        readableListening: false,
        objectMode: false,
        defaultEncoding: 'utf8',
        ranOut: false,
        awaitDrain: 0,
        readingMore: false,
        decoder: null,
        encoding: null },
     readable: false,
     domain: null,
     _events: 
      { error: [Object],
        end: [Object],
        data: [Function],
        readable: [Function],
        close: [Function] },
     _maxListeners: 10,
     socket: 
      { _connecting: false,
        _handle: [Object],
        _readableState: [Object],
        readable: true,
        domain: null,
        _events: [Object],
        _maxListeners: 10,
        _writableState: [Object],
        writable: true,
        allowHalfOpen: true,
        onend: [Function],
        destroyed: false,
        errorEmitted: false,
        bytesRead: 190,
        _bytesDispatched: 0,
        _pendingData: null,
        _pendingEncoding: '',
        server: [Object],
        _idleTimeout: 60000,
        _idleNext: [Object],
        _idlePrev: [Object],
        _idleStart: 1428094579248,
        parser: [Object],
        ondata: [Function],
        _peername: [Object],
        _httpMessage: [Object] },
     connection: 
      { _connecting: false,
        _handle: [Object],
        _readableState: [Object],
        readable: true,
        domain: null,
        _events: [Object],
        _maxListeners: 10,
        _writableState: [Object],
        writable: true,
        allowHalfOpen: true,
        onend: [Function],
        destroyed: false,
        errorEmitted: false,
        bytesRead: 190,
        _bytesDispatched: 0,
        _pendingData: null,
        _pendingEncoding: '',
        server: [Object],
        _idleTimeout: 60000,
        _idleNext: [Object],
        _idlePrev: [Object],
        _idleStart: 1428094579248,
        parser: [Object],
        ondata: [Function],
        _peername: [Object],
        _httpMessage: [Object] },
     httpVersion: '1.1',
     complete: true,
     headers: 
      { host: '10.211.101.57:4444',
        connection: 'Keep-Alive',
        'user-agent': 'Apache-HttpClient/4.3.6 (java 1.5)',
        'accept-encoding': 'gzip,deflate' },
     trailers: {},
     _pendings: [],
     _pendingIndex: 0,
     url: '/grid/api/proxy?id=http://10.211.101.57:5555',
     method: 'GET',
     statusCode: null,
     client: 
      { _connecting: false,
        _handle: [Object],
        _readableState: [Object],
        readable: true,
        domain: null,
        _events: [Object],
        _maxListeners: 10,
        _writableState: [Object],
        writable: true,
        allowHalfOpen: true,
        onend: [Function],
        destroyed: false,
        errorEmitted: false,
        bytesRead: 190,
        _bytesDispatched: 0,
        _pendingData: null,
        _pendingEncoding: '',
        server: [Object],
        _idleTimeout: 60000,
        _idleNext: [Object],
        _idlePrev: [Object],
        _idleStart: 1428094579248,
        parser: [Object],
        ondata: [Function],
        _peername: [Object],
        _httpMessage: [Object] },
     _consuming: true,
     _dumped: false,
     httpVersionMajor: 1,
     httpVersionMinor: 1,
     upgrade: false,
     pipe: [Function],
     addListener: [Function],
     on: [Function],
     pause: [Function],
     resume: [Function],
     read: [Function] },
  nodeRequest: 
   { _readableState: 
      { highWaterMark: 16384,
        buffer: [],
        length: 0,
        pipes: null,
        pipesCount: 0,
        flowing: false,
        ended: true,
        endEmitted: true,
        reading: false,
        calledRead: true,
        sync: false,
        needReadable: true,
        emittedReadable: false,
        readableListening: false,
        objectMode: false,
        defaultEncoding: 'utf8',
        ranOut: false,
        awaitDrain: 0,
        readingMore: false,
        decoder: null,
        encoding: null },
     readable: false,
     domain: null,
     _events: 
      { error: [Object],
        end: [Object],
        data: [Function],
        readable: [Function],
        close: [Function] },
     _maxListeners: 10,
     socket: 
      { _connecting: false,
        _handle: [Object],
        _readableState: [Object],
        readable: true,
        domain: null,
        _events: [Object],
        _maxListeners: 10,
        _writableState: [Object],
        writable: true,
        allowHalfOpen: true,
        onend: [Function],
        destroyed: false,
        errorEmitted: false,
        bytesRead: 190,
        _bytesDispatched: 0,
        _pendingData: null,
        _pendingEncoding: '',
        server: [Object],
        _idleTimeout: 60000,
        _idleNext: [Object],
        _idlePrev: [Object],
        _idleStart: 1428094579248,
        parser: [Object],
        ondata: [Function],
        _peername: [Object],
        _httpMessage: [Object] },
     connection: 
      { _connecting: false,
        _handle: [Object],
        _readableState: [Object],
        readable: true,
        domain: null,
        _events: [Object],
        _maxListeners: 10,
        _writableState: [Object],
        writable: true,
        allowHalfOpen: true,
        onend: [Function],
        destroyed: false,
        errorEmitted: false,
        bytesRead: 190,
        _bytesDispatched: 0,
        _pendingData: null,
        _pendingEncoding: '',
        server: [Object],
        _idleTimeout: 60000,
        _idleNext: [Object],
        _idlePrev: [Object],
        _idleStart: 1428094579248,
        parser: [Object],
        ondata: [Function],
        _peername: [Object],
        _httpMessage: [Object] },
     httpVersion: '1.1',
     complete: true,
     headers: 
      { host: '10.211.101.57:4444',
        connection: 'Keep-Alive',
        'user-agent': 'Apache-HttpClient/4.3.6 (java 1.5)',
        'accept-encoding': 'gzip,deflate' },
     trailers: {},
     _pendings: [],
     _pendingIndex: 0,
     url: '/grid/api/proxy?id=http://10.211.101.57:5555',
     method: 'GET',
     statusCode: null,
     client: 
      { _connecting: false,
        _handle: [Object],
        _readableState: [Object],
        readable: true,
        domain: null,
        _events: [Object],
        _maxListeners: 10,
        _writableState: [Object],
        writable: true,
        allowHalfOpen: true,
        onend: [Function],
        destroyed: false,
        errorEmitted: false,
        bytesRead: 190,
        _bytesDispatched: 0,
        _pendingData: null,
        _pendingEncoding: '',
        server: [Object],
        _idleTimeout: 60000,
        _idleNext: [Object],
        _idlePrev: [Object],
        _idleStart: 1428094579248,
        parser: [Object],
        ondata: [Function],
        _peername: [Object],
        _httpMessage: [Object] },
     _consuming: true,
     _dumped: false,
     httpVersionMajor: 1,
     httpVersionMinor: 1,
     upgrade: false,
     pipe: [Function],
     addListener: [Function],
     on: [Function],
     pause: [Function],
     resume: [Function],
     read: [Function] },
  nodeConnection: 
   { _connecting: false,
     _handle: 
      { fd: 17,
        writeQueueSize: 0,
        owner: [Circular],
        onread: [Function: onread],
        reading: true },
     _readableState: 
      { highWaterMark: 16384,
        buffer: [],
        length: 0,
        pipes: null,
        pipesCount: 0,
        flowing: false,
        ended: false,
        endEmitted: false,
        reading: true,
        calledRead: true,
        sync: false,
        needReadable: true,
        emittedReadable: false,
        readableListening: false,
        objectMode: false,
        defaultEncoding: 'utf8',
        ranOut: false,
        awaitDrain: 0,
        readingMore: false,
        decoder: null,
        encoding: null },
     readable: true,
     domain: null,
     _events: 
      { end: [Object],
        finish: [Function: onSocketFinish],
        _socketEnd: [Function: onSocketEnd],
        drain: [Function: ondrain],
        timeout: [Object],
        error: [Function],
        close: [Object] },
     _maxListeners: 10,
     _writableState: 
      { highWaterMark: 16384,
        objectMode: false,
        needDrain: false,
        ending: false,
        ended: false,
        finished: false,
        decodeStrings: false,
        defaultEncoding: 'utf8',
        length: 0,
        writing: false,
        sync: true,
        bufferProcessing: false,
        onwrite: [Function],
        writecb: null,
        writelen: 0,
        buffer: [] },
     writable: true,
     allowHalfOpen: true,
     onend: [Function],
     destroyed: false,
     errorEmitted: false,
     bytesRead: 190,
     _bytesDispatched: 0,
     _pendingData: null,
     _pendingEncoding: '',
     server: 
      { domain: null,
        _events: [Object],
        _maxListeners: 10,
        _connections: 3,
        connections: [Getter/Setter],
        _handle: [Object],
        _usingSlaves: false,
        _slaves: [],
        allowHalfOpen: true,
        httpAllowHalfOpen: true,
        timeout: 120000,
        destroy: [Function],
        _connectionKey: '6::::4444' },
     _idleTimeout: 60000,
     _idleNext: 
      { _connecting: false,
        _handle: [Object],
        _readableState: [Object],
        readable: true,
        domain: null,
        _events: [Object],
        _maxListeners: 10,
        _writableState: [Object],
        writable: true,
        allowHalfOpen: true,
        onend: [Function],
        destroyed: false,
        errorEmitted: false,
        bytesRead: 336,
        _bytesDispatched: 0,
        _pendingData: null,
        _pendingEncoding: '',
        server: [Object],
        _idleTimeout: 60000,
        _idleNext: [Object],
        _idlePrev: [Circular],
        _idleStart: 1428094576218,
        parser: [Object],
        ondata: [Function],
        _peername: [Object],
        _httpMessage: [Object] },
     _idlePrev: { _idleNext: [Circular], _idlePrev: [Object] },
     _idleStart: 1428094579248,
     parser: 
      { _headers: [],
        _url: '',
        onHeaders: [Function: parserOnHeaders],
        onHeadersComplete: [Function: parserOnHeadersComplete],
        onBody: [Function: parserOnBody],
        onMessageComplete: [Function: parserOnMessageComplete],
        socket: [Circular],
        incoming: [Object],
        maxHeaderPairs: 2000,
        onIncoming: [Function] },
     ondata: [Function],
     _peername: { address: '::ffff:10.211.101.57', family: 'IPv6', port: 54457 },
     _httpMessage: 
      { domain: null,
        _events: [Object],
        _maxListeners: 10,
        output: [],
        outputEncodings: [],
        writable: true,
        _last: false,
        chunkedEncoding: false,
        shouldKeepAlive: true,
        useChunkedEncodingByDefault: true,
        sendDate: true,
        _hasBody: true,
        _trailer: '',
        finished: false,
        _hangupClose: false,
        socket: [Circular],
        connection: [Circular] } },
  query: { id: 'http://10.211.101.57:5555' },
  pathname: '/grid/api/proxy' }
```
Do you have any ideas @arikon ?

Here's the whole log from `selenium-grid.log`:

```
2015-04-07T14:15:06-0400 <info> log.js:54 Hub is listening on http://localhost:4444
2015-04-07T14:15:20-0400 <debug> log.js:60 ::ffff:127.0.0.1:59860 -->     GET /grid/api/hub HTTP/1.1
2015-04-07T14:15:20-0400 <debug> log.js:60 ::ffff:127.0.0.1:59860 <== 200 GET /grid/api/hub HTTP/1.1 -
2015-04-07T14:15:21-0400 <debug> log.js:60 ::ffff:127.0.0.1:59860 -->     POST /grid/register HTTP/1.1
2015-04-07T14:15:21-0400 <debug> log.js:60 Started monitoring for node with ID=http://192.168.0.20:5555; 5000ms interval
2015-04-07T14:15:21-0400 <info> log.js:54 Registered node with ID=http://192.168.0.20:5555
2015-04-07T14:15:21-0400 <debug> log.js:60 ::ffff:127.0.0.1:59860 <== 200 POST /grid/register HTTP/1.1 -
2015-04-07T14:15:26-0400 <debug> log.js:60 Checking status of node with ID=http://192.168.0.20:5555; timeout 2000ms
2015-04-07T14:15:26-0400 <debug> log.js:60 ::ffff:127.0.0.1:59866 -->     GET /grid/api/proxy?id=http://192.168.0.20:5555 HTTP/1.1
2015-04-07T14:15:26-0400 <debug> log.js:60 ::ffff:127.0.0.1:59866 <== 200 GET /grid/api/proxy?id=http://192.168.0.20:5555 HTTP/1.1 -
2015-04-07T14:15:26-0400 <debug> log.js:60 Node with ID=http://192.168.0.20:5555 returned 200 on status check
2015-04-07T14:15:31-0400 <debug> log.js:60 Checking status of node with ID=http://192.168.0.20:5555; timeout 2000ms
2015-04-07T14:15:31-0400 <debug> log.js:60 Node with ID=http://192.168.0.20:5555 returned 200 on status check
2015-04-07T14:15:31-0400 <debug> log.js:60 ::ffff:127.0.0.1:59866 -->     GET /grid/api/proxy?id=http://192.168.0.20:5555 HTTP/1.1
2015-04-07T14:15:31-0400 <debug> log.js:60 ::ffff:127.0.0.1:59866 <== 200 GET /grid/api/proxy?id=http://192.168.0.20:5555 HTTP/1.1 -
2015-04-07T14:15:36-0400 <debug> log.js:60 Checking status of node with ID=http://192.168.0.20:5555; timeout 2000ms
2015-04-07T14:15:36-0400 <debug> log.js:60 Node with ID=http://192.168.0.20:5555 returned 200 on status check
2015-04-07T14:15:36-0400 <debug> log.js:60 ::ffff:127.0.0.1:59866 -->     GET /grid/api/proxy?id=http://192.168.0.20:5555 HTTP/1.1
2015-04-07T14:15:36-0400 <debug> log.js:60 ::ffff:127.0.0.1:59866 <== 200 GET /grid/api/proxy?id=http://192.168.0.20:5555 HTTP/1.1 -
2015-04-07T14:15:41-0400 <debug> log.js:60 Checking status of node with ID=http://192.168.0.20:5555; timeout 2000ms
2015-04-07T14:15:41-0400 <debug> log.js:60 Node with ID=http://192.168.0.20:5555 returned 200 on status check
2015-04-07T14:15:41-0400 <debug> log.js:60 ::ffff:127.0.0.1:59866 -->     GET /grid/api/proxy?id=http://192.168.0.20:5555 HTTP/1.1
2015-04-07T14:15:41-0400 <debug> log.js:60 ::ffff:127.0.0.1:59866 <== 200 GET /grid/api/proxy?id=http://192.168.0.20:5555 HTTP/1.1 -
2015-04-07T14:15:46-0400 <debug> log.js:60 Checking status of node with ID=http://192.168.0.20:5555; timeout 2000ms
2015-04-07T14:15:46-0400 <debug> log.js:60 Node with ID=http://192.168.0.20:5555 returned 200 on status check
2015-04-07T14:15:46-0400 <debug> log.js:60 ::ffff:127.0.0.1:59866 -->     GET /grid/api/proxy?id=http://192.168.0.20:5555 HTTP/1.1
2015-04-07T14:15:46-0400 <debug> log.js:60 ::ffff:127.0.0.1:59866 <== 200 GET /grid/api/proxy?id=http://192.168.0.20:5555 HTTP/1.1 -
2015-04-07T14:15:51-0400 <debug> log.js:60 Checking status of node with ID=http://192.168.0.20:5555; timeout 2000ms
2015-04-07T14:15:51-0400 <debug> log.js:60 Node with ID=http://192.168.0.20:5555 returned 200 on status check
2015-04-07T14:15:51-0400 <debug> log.js:60 ::ffff:127.0.0.1:59866 -->     GET /grid/api/proxy?id=http://192.168.0.20:5555 HTTP/1.1
2015-04-07T14:15:51-0400 <debug> log.js:60 ::ffff:127.0.0.1:59866 <== 200 GET /grid/api/proxy?id=http://192.168.0.20:5555 HTTP/1.1 -
2015-04-07T14:15:56-0400 <debug> log.js:60 Checking status of node with ID=http://192.168.0.20:5555; timeout 2000ms
2015-04-07T14:15:56-0400 <debug> log.js:60 Node with ID=http://192.168.0.20:5555 returned 200 on status check
2015-04-07T14:15:56-0400 <debug> log.js:60 ::ffff:127.0.0.1:59866 -->     GET /grid/api/proxy?id=http://192.168.0.20:5555 HTTP/1.1
2015-04-07T14:15:56-0400 <debug> log.js:60 ::ffff:127.0.0.1:59866 <== 200 GET /grid/api/proxy?id=http://192.168.0.20:5555 HTTP/1.1 -
2015-04-07T14:16:01-0400 <debug> log.js:60 Checking status of node with ID=http://192.168.0.20:5555; timeout 2000ms
2015-04-07T14:16:01-0400 <debug> log.js:60 Node with ID=http://192.168.0.20:5555 returned 200 on status check
2015-04-07T14:16:01-0400 <debug> log.js:60 ::ffff:127.0.0.1:59866 -->     GET /grid/api/proxy?id=http://192.168.0.20:5555 HTTP/1.1
2015-04-07T14:16:01-0400 <debug> log.js:60 ::ffff:127.0.0.1:59866 <== 200 GET /grid/api/proxy?id=http://192.168.0.20:5555 HTTP/1.1 -
2015-04-07T14:16:06-0400 <debug> log.js:60 Checking status of node with ID=http://192.168.0.20:5555; timeout 2000ms
2015-04-07T14:16:06-0400 <debug> log.js:60 Node with ID=http://192.168.0.20:5555 returned 200 on status check
2015-04-07T14:16:06-0400 <debug> log.js:60 ::ffff:127.0.0.1:59866 -->     GET /grid/api/proxy?id=http://192.168.0.20:5555 HTTP/1.1
2015-04-07T14:16:06-0400 <debug> log.js:60 ::ffff:127.0.0.1:59866 <== 200 GET /grid/api/proxy?id=http://192.168.0.20:5555 HTTP/1.1 -
2015-04-07T14:16:11-0400 <debug> log.js:60 Checking status of node with ID=http://192.168.0.20:5555; timeout 2000ms
2015-04-07T14:16:11-0400 <debug> log.js:60 Node with ID=http://192.168.0.20:5555 returned 200 on status check
2015-04-07T14:16:11-0400 <debug> log.js:60 ::ffff:127.0.0.1:59866 -->     GET /grid/api/proxy?id=http://192.168.0.20:5555 HTTP/1.1
2015-04-07T14:16:11-0400 <debug> log.js:60 ::ffff:127.0.0.1:59866 <== 200 GET /grid/api/proxy?id=http://192.168.0.20:5555 HTTP/1.1 -
2015-04-07T14:16:15-0400 <debug> log.js:60 ::ffff:127.0.0.1:59930 -->     POST /wd/hub/session HTTP/1.1
2015-04-07T14:16:15-0400 <debug> log.js:60 Trying to open session for node with ID=http://192.168.0.20:5555
2015-04-07T14:16:15-0400 <debug> log.js:60 Capabilities for slot on node with ID=http://192.168.0.20:5555 do not match
2015-04-07T14:16:15-0400 <debug> log.js:60 * desired capabilities: {"version":"","browserName":"firefox","platform":"ANY","javascriptEnabled":true}
2015-04-07T14:16:15-0400 <debug> log.js:60 * slot capabilities: {"seleniumProtocol":"Selenium","browserName":"*firefox","maxInstances":5,"platform":"MAC"}
2015-04-07T14:16:15-0400 <debug> log.js:60 Capabilities for slot on node with ID=http://192.168.0.20:5555 do not match
2015-04-07T14:16:15-0400 <debug> log.js:60 * desired capabilities: {"version":"","browserName":"firefox","platform":"ANY","javascriptEnabled":true}
2015-04-07T14:16:15-0400 <debug> log.js:60 * slot capabilities: {"seleniumProtocol":"Selenium","browserName":"*firefox","maxInstances":5,"platform":"MAC"}
2015-04-07T14:16:15-0400 <debug> log.js:60 Capabilities for slot on node with ID=http://192.168.0.20:5555 do not match
2015-04-07T14:16:15-0400 <debug> log.js:60 * desired capabilities: {"version":"","browserName":"firefox","platform":"ANY","javascriptEnabled":true}
2015-04-07T14:16:15-0400 <debug> log.js:60 * slot capabilities: {"seleniumProtocol":"Selenium","browserName":"*firefox","maxInstances":5,"platform":"MAC"}
2015-04-07T14:16:15-0400 <debug> log.js:60 Capabilities for slot on node with ID=http://192.168.0.20:5555 do not match
2015-04-07T14:16:15-0400 <debug> log.js:60 * desired capabilities: {"version":"","browserName":"firefox","platform":"ANY","javascriptEnabled":true}
2015-04-07T14:16:15-0400 <debug> log.js:60 * slot capabilities: {"seleniumProtocol":"Selenium","browserName":"*firefox","maxInstances":5,"platform":"MAC"}
2015-04-07T14:16:15-0400 <debug> log.js:60 Capabilities for slot on node with ID=http://192.168.0.20:5555 do not match
2015-04-07T14:16:15-0400 <debug> log.js:60 * desired capabilities: {"version":"","browserName":"firefox","platform":"ANY","javascriptEnabled":true}
2015-04-07T14:16:15-0400 <debug> log.js:60 * slot capabilities: {"seleniumProtocol":"Selenium","browserName":"*firefox","maxInstances":5,"platform":"MAC"}
2015-04-07T14:16:15-0400 <debug> log.js:60 Capabilities for slot on node with ID=http://192.168.0.20:5555 do not match
2015-04-07T14:16:15-0400 <debug> log.js:60 * desired capabilities: {"version":"","browserName":"firefox","platform":"ANY","javascriptEnabled":true}
2015-04-07T14:16:15-0400 <debug> log.js:60 * slot capabilities: {"seleniumProtocol":"Selenium","browserName":"*googlechrome","maxInstances":5,"platform":"MAC"}
2015-04-07T14:16:15-0400 <debug> log.js:60 Capabilities for slot on node with ID=http://192.168.0.20:5555 do not match
2015-04-07T14:16:15-0400 <debug> log.js:60 * desired capabilities: {"version":"","browserName":"firefox","platform":"ANY","javascriptEnabled":true}
2015-04-07T14:16:15-0400 <debug> log.js:60 * slot capabilities: {"seleniumProtocol":"Selenium","browserName":"*googlechrome","maxInstances":5,"platform":"MAC"}
2015-04-07T14:16:15-0400 <debug> log.js:60 Capabilities for slot on node with ID=http://192.168.0.20:5555 do not match
2015-04-07T14:16:15-0400 <debug> log.js:60 * desired capabilities: {"version":"","browserName":"firefox","platform":"ANY","javascriptEnabled":true}
2015-04-07T14:16:15-0400 <debug> log.js:60 * slot capabilities: {"seleniumProtocol":"Selenium","browserName":"*googlechrome","maxInstances":5,"platform":"MAC"}
2015-04-07T14:16:15-0400 <debug> log.js:60 Capabilities for slot on node with ID=http://192.168.0.20:5555 do not match
2015-04-07T14:16:15-0400 <debug> log.js:60 * desired capabilities: {"version":"","browserName":"firefox","platform":"ANY","javascriptEnabled":true}
2015-04-07T14:16:15-0400 <debug> log.js:60 * slot capabilities: {"seleniumProtocol":"Selenium","browserName":"*googlechrome","maxInstances":5,"platform":"MAC"}
2015-04-07T14:16:15-0400 <debug> log.js:60 Capabilities for slot on node with ID=http://192.168.0.20:5555 do not match
2015-04-07T14:16:15-0400 <debug> log.js:60 * desired capabilities: {"version":"","browserName":"firefox","platform":"ANY","javascriptEnabled":true}
2015-04-07T14:16:15-0400 <debug> log.js:60 * slot capabilities: {"seleniumProtocol":"Selenium","browserName":"*googlechrome","maxInstances":5,"platform":"MAC"}
2015-04-07T14:16:15-0400 <debug> log.js:60 Capabilities for slot on node with ID=http://192.168.0.20:5555 do not match
2015-04-07T14:16:15-0400 <debug> log.js:60 * desired capabilities: {"version":"","browserName":"firefox","platform":"ANY","javascriptEnabled":true}
2015-04-07T14:16:15-0400 <debug> log.js:60 * slot capabilities: {"seleniumProtocol":"Selenium","browserName":"*iexplore","maxInstances":1,"platform":"MAC"}
2015-04-07T14:16:16-0400 <debug> log.js:60 Checking status of node with ID=http://192.168.0.20:5555; timeout 2000ms
2015-04-07T14:16:16-0400 <debug> log.js:60 Node with ID=http://192.168.0.20:5555 returned 200 on status check
2015-04-07T14:16:16-0400 <debug> log.js:60 ::ffff:127.0.0.1:59866 -->     GET /grid/api/proxy?id=http://192.168.0.20:5555 HTTP/1.1
2015-04-07T14:16:16-0400 <debug> log.js:60 ::ffff:127.0.0.1:59866 <== 200 GET /grid/api/proxy?id=http://192.168.0.20:5555 HTTP/1.1 -
2015-04-07T14:16:21-0400 <debug> log.js:60 Checking status of node with ID=http://192.168.0.20:5555; timeout 2000ms
2015-04-07T14:16:21-0400 <debug> log.js:60 Node with ID=http://192.168.0.20:5555 returned 200 on status check
2015-04-07T14:16:21-0400 <debug> log.js:60 ::ffff:127.0.0.1:59866 -->     GET /grid/api/proxy?id=http://192.168.0.20:5555 HTTP/1.1
2015-04-07T14:16:21-0400 <debug> log.js:60 ::ffff:127.0.0.1:59866 <== 200 GET /grid/api/proxy?id=http://192.168.0.20:5555 HTTP/1.1 -
2015-04-07T14:16:26-0400 <debug> log.js:60 Checking status of node with ID=http://192.168.0.20:5555; timeout 2000ms
2015-04-07T14:16:26-0400 <debug> log.js:60 Node with ID=http://192.168.0.20:5555 returned 200 on status check
2015-04-07T14:16:26-0400 <debug> log.js:60 ::ffff:127.0.0.1:59866 -->     GET /grid/api/proxy?id=http://192.168.0.20:5555 HTTP/1.1
2015-04-07T14:16:26-0400 <debug> log.js:60 ::ffff:127.0.0.1:59866 <== 200 GET /grid/api/proxy?id=http://192.168.0.20:5555 HTTP/1.1 -
2015-04-07T14:16:31-0400 <debug> log.js:60 Checking status of node with ID=http://192.168.0.20:5555; timeout 2000ms
2015-04-07T14:16:31-0400 <debug> log.js:60 Node with ID=http://192.168.0.20:5555 returned 200 on status check
2015-04-07T14:16:31-0400 <debug> log.js:60 ::ffff:127.0.0.1:59866 -->     GET /grid/api/proxy?id=http://192.168.0.20:5555 HTTP/1.1
2015-04-07T14:16:31-0400 <debug> log.js:60 ::ffff:127.0.0.1:59866 <== 200 GET /grid/api/proxy?id=http://192.168.0.20:5555 HTTP/1.1 -
2015-04-07T14:16:36-0400 <debug> log.js:60 Checking status of node with ID=http://192.168.0.20:5555; timeout 2000ms
2015-04-07T14:16:36-0400 <debug> log.js:60 Node with ID=http://192.168.0.20:5555 returned 200 on status check
2015-04-07T14:16:36-0400 <debug> log.js:60 ::ffff:127.0.0.1:59866 -->     GET /grid/api/proxy?id=http://192.168.0.20:5555 HTTP/1.1
2015-04-07T14:16:36-0400 <debug> log.js:60 ::ffff:127.0.0.1:59866 <== 200 GET /grid/api/proxy?id=http://192.168.0.20:5555 HTTP/1.1 -
2015-04-07T14:16:41-0400 <debug> log.js:60 Checking status of node with ID=http://192.168.0.20:5555; timeout 2000ms
2015-04-07T14:16:41-0400 <debug> log.js:60 Node with ID=http://192.168.0.20:5555 returned 200 on status check
2015-04-07T14:16:41-0400 <debug> log.js:60 ::ffff:127.0.0.1:59866 -->     GET /grid/api/proxy?id=http://192.168.0.20:5555 HTTP/1.1
2015-04-07T14:16:41-0400 <debug> log.js:60 ::ffff:127.0.0.1:59866 <== 200 GET /grid/api/proxy?id=http://192.168.0.20:5555 HTTP/1.1 -
2015-04-07T14:16:46-0400 <debug> log.js:60 Checking status of node with ID=http://192.168.0.20:5555; timeout 2000ms
2015-04-07T14:16:46-0400 <debug> log.js:60 Node with ID=http://192.168.0.20:5555 returned 200 on status check
2015-04-07T14:16:46-0400 <debug> log.js:60 ::ffff:127.0.0.1:59866 -->     GET /grid/api/proxy?id=http://192.168.0.20:5555 HTTP/1.1
2015-04-07T14:16:46-0400 <debug> log.js:60 ::ffff:127.0.0.1:59866 <== 200 GET /grid/api/proxy?id=http://192.168.0.20:5555 HTTP/1.1 -
2015-04-07T14:16:51-0400 <debug> log.js:60 Checking status of node with ID=http://192.168.0.20:5555; timeout 2000ms
2015-04-07T14:16:51-0400 <debug> log.js:60 Node with ID=http://192.168.0.20:5555 returned 200 on status check
2015-04-07T14:16:51-0400 <debug> log.js:60 ::ffff:127.0.0.1:59866 -->     GET /grid/api/proxy?id=http://192.168.0.20:5555 HTTP/1.1
2015-04-07T14:16:51-0400 <debug> log.js:60 ::ffff:127.0.0.1:59866 <== 200 GET /grid/api/proxy?id=http://192.168.0.20:5555 HTTP/1.1 -
2015-04-07T14:16:56-0400 <debug> log.js:60 Checking status of node with ID=http://192.168.0.20:5555; timeout 2000ms
2015-04-07T14:16:56-0400 <debug> log.js:60 Node with ID=http://192.168.0.20:5555 returned 200 on status check
2015-04-07T14:16:56-0400 <debug> log.js:60 ::ffff:127.0.0.1:59866 -->     GET /grid/api/proxy?id=http://192.168.0.20:5555 HTTP/1.1
2015-04-07T14:16:56-0400 <debug> log.js:60 ::ffff:127.0.0.1:59866 <== 200 GET /grid/api/proxy?id=http://192.168.0.20:5555 HTTP/1.1 -
2015-04-07T14:17:01-0400 <debug> log.js:60 Checking status of node with ID=http://192.168.0.20:5555; timeout 2000ms
2015-04-07T14:17:01-0400 <debug> log.js:60 Node with ID=http://192.168.0.20:5555 returned 200 on status check
2015-04-07T14:17:01-0400 <debug> log.js:60 ::ffff:127.0.0.1:59866 -->     GET /grid/api/proxy?id=http://192.168.0.20:5555 HTTP/1.1
2015-04-07T14:17:01-0400 <debug> log.js:60 ::ffff:127.0.0.1:59866 <== 200 GET /grid/api/proxy?id=http://192.168.0.20:5555 HTTP/1.1 -
2015-04-07T14:17:06-0400 <debug> log.js:60 Checking status of node with ID=http://192.168.0.20:5555; timeout 2000ms
2015-04-07T14:17:06-0400 <debug> log.js:60 Node with ID=http://192.168.0.20:5555 returned 200 on status check
2015-04-07T14:17:06-0400 <debug> log.js:60 ::ffff:127.0.0.1:59866 -->     GET /grid/api/proxy?id=http://192.168.0.20:5555 HTTP/1.1
2015-04-07T14:17:06-0400 <debug> log.js:60 ::ffff:127.0.0.1:59866 <== 200 GET /grid/api/proxy?id=http://192.168.0.20:5555 HTTP/1.1 -
2015-04-07T14:17:11-0400 <debug> log.js:60 Checking status of node with ID=http://192.168.0.20:5555; timeout 2000ms
2015-04-07T14:17:11-0400 <debug> log.js:60 Node with ID=http://192.168.0.20:5555 returned 200 on status check
2015-04-07T14:17:11-0400 <debug> log.js:60 ::ffff:127.0.0.1:59866 -->     GET /grid/api/proxy?id=http://192.168.0.20:5555 HTTP/1.1
2015-04-07T14:17:11-0400 <debug> log.js:60 ::ffff:127.0.0.1:59866 <== 200 GET /grid/api/proxy?id=http://192.168.0.20:5555 HTTP/1.1 -
2015-04-07T14:17:15-0400 <debug> log.js:60 Error occured during session opening on node with ID=http://192.168.0.20:5555
2015-04-07T14:17:15-0400 <debug> log.js:60 Node responded with an error:
<html>
<head>
<title>Error 500 Read timed out</title>
</head>
<body>
<h2>HTTP ERROR: 500</h2><pre>Read timed out</pre>
<p>RequestURI=/wd/hub/session</p>
<p><i><small><a href="http://jetty.mortbay.org">Powered by Jetty://</a></small></i></p>




















</body>
</html>

2015-04-07T14:17:15-0400 <warn> log.js:66 Could not create session because of error:
Error: Node responded with an error:
<html>
<head>
<title>Error 500 Read timed out</title>
</head>
<body>
<h2>HTTP ERROR: 500</h2><pre>Read timed out</pre>
<p>RequestURI=/wd/hub/session</p>
<p><i><small><a href="http://jetty.mortbay.org">Powered by Jetty://</a></small></i></p>




















</body>
</html>
    at /Users/Laurent/Downloads/selenodium-grid-dev/lib/driver/CommonImpl.js:54:41
From previous event:
    at /Users/Laurent/Downloads/selenodium-grid-dev/lib/driver/CommonImpl.js:51:22
From previous event:
    at module.exports.inherit.getNewSessionFromNode (/Users/Laurent/Downloads/selenodium-grid-dev/lib/driver/CommonImpl.js:40:19)
    at inherit.getNewSession (/Users/Laurent/Downloads/selenodium-grid-dev/lib/registry/NodeSlot.js:54:33)
    at /Users/Laurent/Downloads/selenodium-grid-dev/lib/registry/Node.js:113:25
    at /Users/Laurent/Downloads/selenodium-grid-dev/lib/q-async.js:22:14
2015-04-07T14:17:15-0400 <debug> log.js:60 Trying to open session for node with ID=http://192.168.0.20:5555
2015-04-07T14:17:15-0400 <debug> log.js:60 Capabilities for slot on node with ID=http://192.168.0.20:5555 do not match
2015-04-07T14:17:15-0400 <debug> log.js:60 * desired capabilities: {"version":"","browserName":"firefox","platform":"ANY","javascriptEnabled":true}
2015-04-07T14:17:15-0400 <debug> log.js:60 * slot capabilities: {"seleniumProtocol":"Selenium","browserName":"*firefox","maxInstances":5,"platform":"MAC"}
2015-04-07T14:17:15-0400 <debug> log.js:60 Capabilities for slot on node with ID=http://192.168.0.20:5555 do not match
2015-04-07T14:17:15-0400 <debug> log.js:60 * desired capabilities: {"version":"","browserName":"firefox","platform":"ANY","javascriptEnabled":true}
2015-04-07T14:17:15-0400 <debug> log.js:60 * slot capabilities: {"seleniumProtocol":"Selenium","browserName":"*firefox","maxInstances":5,"platform":"MAC"}
2015-04-07T14:17:15-0400 <debug> log.js:60 Capabilities for slot on node with ID=http://192.168.0.20:5555 do not match
2015-04-07T14:17:15-0400 <debug> log.js:60 * desired capabilities: {"version":"","browserName":"firefox","platform":"ANY","javascriptEnabled":true}
2015-04-07T14:17:15-0400 <debug> log.js:60 * slot capabilities: {"seleniumProtocol":"Selenium","browserName":"*firefox","maxInstances":5,"platform":"MAC"}
2015-04-07T14:17:15-0400 <debug> log.js:60 Capabilities for slot on node with ID=http://192.168.0.20:5555 do not match
2015-04-07T14:17:15-0400 <debug> log.js:60 * desired capabilities: {"version":"","browserName":"firefox","platform":"ANY","javascriptEnabled":true}
2015-04-07T14:17:15-0400 <debug> log.js:60 * slot capabilities: {"seleniumProtocol":"Selenium","browserName":"*firefox","maxInstances":5,"platform":"MAC"}
2015-04-07T14:17:15-0400 <debug> log.js:60 Capabilities for slot on node with ID=http://192.168.0.20:5555 do not match
2015-04-07T14:17:15-0400 <debug> log.js:60 * desired capabilities: {"version":"","browserName":"firefox","platform":"ANY","javascriptEnabled":true}
2015-04-07T14:17:15-0400 <debug> log.js:60 * slot capabilities: {"seleniumProtocol":"Selenium","browserName":"*firefox","maxInstances":5,"platform":"MAC"}
2015-04-07T14:17:15-0400 <debug> log.js:60 Capabilities for slot on node with ID=http://192.168.0.20:5555 do not match
2015-04-07T14:17:15-0400 <debug> log.js:60 * desired capabilities: {"version":"","browserName":"firefox","platform":"ANY","javascriptEnabled":true}
2015-04-07T14:17:15-0400 <debug> log.js:60 * slot capabilities: {"seleniumProtocol":"Selenium","browserName":"*googlechrome","maxInstances":5,"platform":"MAC"}
2015-04-07T14:17:15-0400 <debug> log.js:60 Capabilities for slot on node with ID=http://192.168.0.20:5555 do not match
2015-04-07T14:17:15-0400 <debug> log.js:60 * desired capabilities: {"version":"","browserName":"firefox","platform":"ANY","javascriptEnabled":true}
2015-04-07T14:17:15-0400 <debug> log.js:60 * slot capabilities: {"seleniumProtocol":"Selenium","browserName":"*googlechrome","maxInstances":5,"platform":"MAC"}
2015-04-07T14:17:15-0400 <debug> log.js:60 Capabilities for slot on node with ID=http://192.168.0.20:5555 do not match
2015-04-07T14:17:15-0400 <debug> log.js:60 * desired capabilities: {"version":"","browserName":"firefox","platform":"ANY","javascriptEnabled":true}
2015-04-07T14:17:15-0400 <debug> log.js:60 * slot capabilities: {"seleniumProtocol":"Selenium","browserName":"*googlechrome","maxInstances":5,"platform":"MAC"}
2015-04-07T14:17:15-0400 <debug> log.js:60 Capabilities for slot on node with ID=http://192.168.0.20:5555 do not match
2015-04-07T14:17:15-0400 <debug> log.js:60 * desired capabilities: {"version":"","browserName":"firefox","platform":"ANY","javascriptEnabled":true}
2015-04-07T14:17:15-0400 <debug> log.js:60 * slot capabilities: {"seleniumProtocol":"Selenium","browserName":"*googlechrome","maxInstances":5,"platform":"MAC"}
2015-04-07T14:17:15-0400 <debug> log.js:60 Capabilities for slot on node with ID=http://192.168.0.20:5555 do not match
2015-04-07T14:17:15-0400 <debug> log.js:60 * desired capabilities: {"version":"","browserName":"firefox","platform":"ANY","javascriptEnabled":true}
2015-04-07T14:17:15-0400 <debug> log.js:60 * slot capabilities: {"seleniumProtocol":"Selenium","browserName":"*googlechrome","maxInstances":5,"platform":"MAC"}
2015-04-07T14:17:15-0400 <debug> log.js:60 Capabilities for slot on node with ID=http://192.168.0.20:5555 do not match
2015-04-07T14:17:15-0400 <debug> log.js:60 * desired capabilities: {"version":"","browserName":"firefox","platform":"ANY","javascriptEnabled":true}
2015-04-07T14:17:15-0400 <debug> log.js:60 * slot capabilities: {"seleniumProtocol":"Selenium","browserName":"*iexplore","maxInstances":1,"platform":"MAC"}
2015-04-07T14:17:16-0400 <debug> log.js:60 Checking status of node with ID=http://192.168.0.20:5555; timeout 2000ms
2015-04-07T14:17:16-0400 <debug> log.js:60 Node with ID=http://192.168.0.20:5555 returned 200 on status check
2015-04-07T14:17:16-0400 <debug> log.js:60 ::ffff:127.0.0.1:59866 -->     GET /grid/api/proxy?id=http://192.168.0.20:5555 HTTP/1.1
2015-04-07T14:17:16-0400 <debug> log.js:60 ::ffff:127.0.0.1:59866 <== 200 GET /grid/api/proxy?id=http://192.168.0.20:5555 HTTP/1.1 -
2015-04-07T14:17:21-0400 <debug> log.js:60 Checking status of node with ID=http://192.168.0.20:5555; timeout 2000ms
2015-04-07T14:17:21-0400 <debug> log.js:60 Node with ID=http://192.168.0.20:5555 returned 200 on status check
2015-04-07T14:17:21-0400 <debug> log.js:60 ::ffff:127.0.0.1:59866 -->     GET /grid/api/proxy?id=http://192.168.0.20:5555 HTTP/1.1
2015-04-07T14:17:21-0400 <debug> log.js:60 ::ffff:127.0.0.1:59866 <== 200 GET /grid/api/proxy?id=http://192.168.0.20:5555 HTTP/1.1 -
2015-04-07T14:17:26-0400 <debug> log.js:60 Checking status of node with ID=http://192.168.0.20:5555; timeout 2000ms
2015-04-07T14:17:26-0400 <debug> log.js:60 Node with ID=http://192.168.0.20:5555 returned 200 on status check
2015-04-07T14:17:26-0400 <debug> log.js:60 ::ffff:127.0.0.1:59866 -->     GET /grid/api/proxy?id=http://192.168.0.20:5555 HTTP/1.1
2015-04-07T14:17:26-0400 <debug> log.js:60 ::ffff:127.0.0.1:59866 <== 200 GET /grid/api/proxy?id=http://192.168.0.20:5555 HTTP/1.1 -
2015-04-07T14:17:31-0400 <debug> log.js:60 Checking status of node with ID=http://192.168.0.20:5555; timeout 2000ms
2015-04-07T14:17:31-0400 <debug> log.js:60 Node with ID=http://192.168.0.20:5555 returned 200 on status check
2015-04-07T14:17:31-0400 <debug> log.js:60 ::ffff:127.0.0.1:59866 -->     GET /grid/api/proxy?id=http://192.168.0.20:5555 HTTP/1.1
2015-04-07T14:17:31-0400 <debug> log.js:60 ::ffff:127.0.0.1:59866 <== 200 GET /grid/api/proxy?id=http://192.168.0.20:5555 HTTP/1.1 -
2015-04-07T14:17:36-0400 <debug> log.js:60 Checking status of node with ID=http://192.168.0.20:5555; timeout 2000ms
2015-04-07T14:17:36-0400 <debug> log.js:60 Node with ID=http://192.168.0.20:5555 returned 200 on status check
2015-04-07T14:17:36-0400 <debug> log.js:60 ::ffff:127.0.0.1:59866 -->     GET /grid/api/proxy?id=http://192.168.0.20:5555 HTTP/1.1
2015-04-07T14:17:36-0400 <debug> log.js:60 ::ffff:127.0.0.1:59866 <== 200 GET /grid/api/proxy?id=http://192.168.0.20:5555 HTTP/1.1 -
2015-04-07T14:17:41-0400 <debug> log.js:60 Checking status of node with ID=http://192.168.0.20:5555; timeout 2000ms
2015-04-07T14:17:41-0400 <debug> log.js:60 Node with ID=http://192.168.0.20:5555 returned 200 on status check
2015-04-07T14:17:41-0400 <debug> log.js:60 ::ffff:127.0.0.1:59866 -->     GET /grid/api/proxy?id=http://192.168.0.20:5555 HTTP/1.1
2015-04-07T14:17:41-0400 <debug> log.js:60 ::ffff:127.0.0.1:59866 <== 200 GET /grid/api/proxy?id=http://192.168.0.20:5555 HTTP/1.1 -
2015-04-07T14:17:46-0400 <debug> log.js:60 Checking status of node with ID=http://192.168.0.20:5555; timeout 2000ms
2015-04-07T14:17:46-0400 <debug> log.js:60 Node with ID=http://192.168.0.20:5555 returned 200 on status check
```

Until you submit another request to selenodium, the GET requests for node status lookup loop as you would expect. Interestingly, it looks like one of the reasons why a `session` cannot be instantiated and the request eventually times out is because the `desiredCapabilities` do not match, despite the fact that they should.
In case you need them, here is the error trace for the job I sent to `selenodium`:

```
Laurents-iMac:Desktop Laurent$ python3 test.py
Traceback (most recent call last):
  File "test.py", line 7, in <module>
    desired_capabilities=DesiredCapabilities.FIREFOX)
  File "/Library/Frameworks/Python.framework/Versions/3.4/lib/python3.4/site-packages/selenium/webdriver/remote/webdriver.py", line 73, in __init__
    self.start_session(desired_capabilities, browser_profile)
  File "/Library/Frameworks/Python.framework/Versions/3.4/lib/python3.4/site-packages/selenium/webdriver/remote/webdriver.py", line 121, in start_session
    'desiredCapabilities': desired_capabilities,
  File "/Library/Frameworks/Python.framework/Versions/3.4/lib/python3.4/site-packages/selenium/webdriver/remote/webdriver.py", line 173, in execute
    self.error_handler.check_response(response)
  File "/Library/Frameworks/Python.framework/Versions/3.4/lib/python3.4/site-packages/selenium/webdriver/remote/errorhandler.py", line 136, in check_response
    raise exception_class(value)
selenium.common.exceptions.WebDriverException: Message: Error: Socket timed out after 60 seconds
```

As well as the job itself:

``` python
from selenium import webdriver
from selenium.webdriver.common.keys import Keys
from selenium.webdriver.common.desired_capabilities import DesiredCapabilities

driver = webdriver.Remote(
   command_executor='http://localhost:4444/wd/hub',
   desired_capabilities=DesiredCapabilities.FIREFOX)
driver.get("http://www.google.com")
#html = driver.execute_script('return document.body.innerHTML;')
html = driver.page_source
print (html)
driver.close()
```

which uses the Python wrapper for Selenium WebDriver. The selenium node which was started using selenium's standalone server remains however completely unchanged:

```
Laurents-iMac:Desktop Laurent$ java -jar selenium-server-standalone-2.44.0.jar -role node -hub http://localhost:4444/grid.register
14:15:20.689 INFO - Launching a selenium grid node
14:15:20.945 INFO - Java: Oracle Corporation 25.25-b02
14:15:20.945 INFO - OS: Mac OS X 10.10 x86_64
14:15:20.949 INFO - v2.44.0, with Core v2.44.0. Built from revision 76d78cf
14:15:20.981 INFO - Default driver org.openqa.selenium.ie.InternetExplorerDriver registration is skipped: registration capabilities Capabilities [{ensureCleanSession=true, browserName=internet explorer, version=, platform=WINDOWS}] does not match with current platform: MAC
14:15:20.996 INFO - RemoteWebDriver instances should connect to: http://127.0.0.1:5555/wd/hub
14:15:20.997 INFO - Version Jetty/5.1.x
14:15:20.998 INFO - Started HttpContext[/selenium-server,/selenium-server]
14:15:21.015 INFO - Started org.openqa.jetty.jetty.servlet.ServletHandler@319b92f3
14:15:21.015 INFO - Started HttpContext[/wd,/wd]
14:15:21.015 INFO - Started HttpContext[/selenium-server/driver,/selenium-server/driver]
14:15:21.015 INFO - Started HttpContext[/,/]
14:15:21.017 INFO - Started SocketListener on 0.0.0.0:5555
14:15:21.017 INFO - Started org.openqa.jetty.jetty.Server@4fe3c938
14:15:21.034 INFO - using the json request : {"class":"org.openqa.grid.common.RegistrationRequest","configuration":{"proxy":"org.openqa.grid.selenium.proxy.DefaultRemoteProxy","role":"node","hub":"http://localhost:4444/grid.register","port":5555,"remoteHost":"http://192.168.0.20:5555","host":"192.168.0.20","maxSession":5,"hubHost":"localhost","registerCycle":5000,"hubPort":4444,"url":"http://192.168.0.20:5555","register":true},"capabilities":[{"seleniumProtocol":"Selenium","browserName":"*firefox","maxInstances":5,"platform":"MAC"},{"seleniumProtocol":"Selenium","browserName":"*googlechrome","maxInstances":5,"platform":"MAC"},{"seleniumProtocol":"Selenium","browserName":"*iexplore","maxInstances":1,"platform":"MAC"},{"seleniumProtocol":"WebDriver","browserName":"firefox","maxInstances":5,"platform":"MAC"},{"seleniumProtocol":"WebDriver","browserName":"chrome","maxInstances":5,"platform":"MAC"},{"seleniumProtocol":"WebDriver","browserName":"internet explorer","maxInstances":1,"platform":"MAC"}]}
14:15:21.035 INFO - Starting auto register thread. Will try to register every 5000 ms.
14:15:21.035 INFO - Registering the node to hub :http://localhost:4444/grid/register
^C14:22:54.983 INFO - Shutting down...
```

--- a/lib/http-apps.js
+++ b/lib/http-apps.js
@@ -48,7 +48,9 @@ function processJsonBody(obj) {
             if (contentType && contentType.indexOf('json') > -1) {
                 return obj.body.read()
                     .then(function(body) {
-                        obj.data = JSON.parse(body);
+                        if (body.length) {
+                            obj.data = JSON.parse(body);
+                        }
                         return obj;
                     });
             }
@@ -76,7 +78,9 @@ function processUrlEncodedBody(obj) {
             if (contentType && contentType.indexOf('x-www-form-urlencoded') > -1) {
                 return obj.body.read()
                     .then(function(body) {
-                        obj.data = qs.parse(body.toString());
+                        if (body.length) {
+                            obj.data = qs.parse(body.toString());
+                        }
                         return obj;
                     });
             }

--- a/lib/proxy.js
+++ b/lib/proxy.js
@@ -2,6 +2,7 @@
 
 var q = require('q'),
     http = require('q-io/http'),
+    qs = require('qs'),
     log = require('./log');
 
 function proxy(req, node, retryCounter) {
@@ -15,9 +16,20 @@ function proxy(req, node, retryCounter) {
             port: node.port,
             method: req.method,
             path: req.path,
-            headers: req.headers,
-            body: req.data ? [JSON.stringify(req.data)] : req.body
+            headers: {}
         };
+
+    var contentType = req.headers && req.headers['content-type'];
+    if (req.data && contentType) {
+        opts.headers['content-type'] = contentType;
+        if (contentType.indexOf('json') > -1) {
+            opts.body = [JSON.stringify(req.data)];
+        } else if (contentType.indexOf('x-www-form-urlencoded') > -1) {
+            opts.body = [qs.stringify(req.data)];
+        }
+    } else {
+        opts.body = req.body;
+    }
 
     // use separate pool (agent) for each node
     if (typeof node.getAgent === 'function') {

